hibernate-core/src/main/java/org/hibernate/query/sql/internal/ResultSetMappingProcessor.java

```diff
@@ -322,7 +323,7 @@ private void processScalarReturn(NativeQuery.ReturnProperty typeReturn) {
 //		scalarTypes.add( typeReturn.getType() );
 	}
 
-	private void processRootReturn(NativeQuery.RootReturn rootReturn) {
+	private void processRootReturn(NativeQuery.EntityReturn rootReturn) {
```

Comment:
Maybe you should rename the method to `processEntityReturn` and also the variables that refer to such a type accordingly to `entityReturn`